drivers/sensor/explorir_m/explorir_m.c

```diff
@@ -262,6 +264,48 @@ static int explorir_m_uart_transceive(const struct device *dev, char type, struc
 	return rc;
 }
 
+static int explorir_m_calibrate(const struct device *dev, struct sensor_value *val)
+{
+	struct explorir_m_data *data = dev->data;
+	struct sensor_value original;
+	struct sensor_value tmp;
+	int restore_rc;
+	int rc;
+
+	/* Prevent sensor interaction while using calibration filter value */
+	k_mutex_lock(&data->uart_mutex, K_FOREVER);
+
+	rc = explorir_m_uart_transceive(dev, EXPLORIR_M_GET_FILTER_CHAR, &original,
+					EXPLORIR_M_SET_NONE);
+	if (rc != 0) {
+		goto unlock;
+	}
+
+	/*
+	 * From datasheet section "Zero point setting":
+	 * To improve zeroing accuracy, the recommended digital filter setting is 32.
+	 */
+	tmp.val1 = 32;
+	rc = explorir_m_uart_transceive(dev, EXPLORIR_M_SET_FILTER_CHAR, &tmp,
+					EXPLORIR_M_SET_VAL_ONE);
+	if (rc == 0) {
+		tmp.val1 = val->val1 / data->filtered;
```

Comment:
Can you document what `val` is? What units is it in?

Reply:
Added comment above the function.

Reply:
Can `data->filtered` ever be zero?

Reply:
Looking at the datasheet no, it seems to be a multipier for `EXPLORIR_M_CO2_FILTERED_CHAR`.

Also, I think you misunderstood me. I want to know what `val` is, the comment seems to be about what `tmp` is and why it's 32.

Reply:
That comment was there before. The comment i added is [above the function](https://github.com/zephyrproject-rtos/zephyr/compare/a698753bc8cf74cd2582e7a28e685bdcc6423969..3f7cd9c07e4e81c277c32833293e6bfcc942eb41)

Reply:
> Can data->filtered ever be zero?

It looks like it can be either 10 or 100. 0 would not make sense.

I guess noise changing a single bit could potentially change a '1' to a '0' making it read e.g. "00" or "000".
It is not something I have experience but in theory it could be an issue.